### PR TITLE
Added Block MPR reports for icds

### PR DESCRIPTION
**Pull request by lwyszomi**
@NoahCarnahan

### CODE REVIEW

File: custom/icds_reports/sqldata.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    @property
+    def sources(self):
+        with open(os.path.join(os.path.dirname(__file__), 'resources/block_mpr.json')) as f:
+            return json.loads(f.read())[self.slug]
+
+    def custom_data(self, count=False):
+        data = {}
+        for source in self.sources['data_source']:
+            if not count:
+                data.update(ICDSData(source=source, config=self.config).data.get(self.config['location_id'],  {}))
```
Comment (dimagimon) [inline custom/icds_reports/sqldata.py]: E241 multiple spaces after ','

File: custom/icds_reports/sqldata.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                        x = custom_data.get(cell[0], 0)
+                        y = custom_data.get(cell[1], 0)
+                        row_data.append(x + y)
+                    else:
+                        row_data.append(custom_data.get(cell, cell if cell == '--' or idx in [0, 1] else 0))
+                rows.append(row_data)
+            return rows
+
+
+    @property
```
Comment (dimagimon) [inline custom/icds_reports/sqldata.py]: E303 too many blank lines (2)

**Comment by NoahCarnahan** [comment]
Thanks @lwyszomi, and sorry for the delay. I'll review this in the next 24 hours.

**Comment by NoahCarnahan** [comment]
Hey @lwyszomi ,
This looks like a good start.

A few things:
We actually want the configuration file (`block_mpr.json`) to specify `ReportConfiguration`s, not `DataSourceConfiguration`s (and therefore you won't need to specify `"group_by"` or `"agg_fun"` properties, since those will already be configured in the ReportConfiguration).

Also, it would be good to replace the `ICDSData` class with the existing `ConfigurableReportDataSource`. You can instantiate these with `ReportFactory.from_spec(report_configuration)`.

Thanks

**Comment by lwyszomi** [comment]
@NoahCarnahan 

Can we set up the meeting to discuss this, because I noticed some problems in solution which you propose. I'm available from Monday to Friday between 8am-5pm GMT. So you can ping me (skype ID: lwyszomi) when you have a time.

**Comment by lwyszomi** [comment]
@NoahCarnahan 
I'm available between 8am-4pm GMT.

**Comment by lwyszomi** [comment]
@NoahCarnahan 
Now we using ReportFactory to get data.

**Comment by NoahCarnahan** [comment]
Looks good, thanks!

**Pull request merged**